batch.go

```diff
@@ -142,6 +140,8 @@ func (b *Batch) Delete(key []byte) error {
 	}
 
 	b.mu.Lock()
+	defer b.mu.Unlock()
```

Comment:
no need to use defer since the following code is a very common operation without any errors.